tests/framework/microvm.py

```diff
@@ -144,18 +144,31 @@ def __init__(
         # External clone/exec tool, because Python can't into clone
         self.bin_cloner_path = bin_cloner_path
 
+        # Flag checked in destructor to see abnormal signal-induced crashes.
+        self.expect_kill_by_signal = False
+
     def kill(self):
         """All clean up associated with this microVM should go here."""
         # pylint: disable=subprocess-run-check
         if self.logging_thread is not None:
             self.logging_thread.stop()
 
+        if self.expect_kill_by_signal is False and \
+                "Shutting down VM after intercepting signal" in self.log_data:
+            # Too late to assert at this point, pytest will still report the
```

Comment:
I think that before the test is marked as passed/failed, we can add a check in `scheduler.py` that takes a look at what microvms are running and check for problems. This way we can forcibly fail the test if something unexpected happens and the test is marked as passed.

Reply:
> What I'm proposing is slightly more complex, so we can go down the path from this PR and track the improvement as an issue. WDYT?

++